src/settings/StaffSlips/components/ViewSections/StaffSlipTemplateContentSection/StaffSlipTemplateContentSection.test.js

```diff
@@ -0,0 +1,137 @@
+  children,
+  'data-testid': testId,
+  onClick,
+}) => (
+  // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+  <div
+    data-testid={testId}
+    onClick={onClick}
+  >
+    {children}
+  </div>
+)));
+
+describe('StaffSlipTemplateContentSection', () => {
+  const testIds = {
+    staffSlipsDisplayCol: 'staffSlipsDisplayCol',
+    staffSlipsPreviewButton: 'staffSlipsPreviewButton',
+    emailTemplateCol: 'emailTemplateCol',
+    previewModal: 'previewModal',
+  };
+
+  const testInitialValues = {
+    name: 'testNameValue',
+    description: 'testDescriptionValue',
+    template: 'testTemplateValue',
+  };
+
+  const labelIds = {
+    staffSlipsName: 'ui-circulation.settings.staffSlips.name',
+    staffSlipsDescription: 'ui-circulation.settings.staffSlips.description',
+    staffSlipsDisplay: 'ui-circulation.settings.staffSlips.display',
+    staffSlipsPreview: 'ui-circulation.settings.staffSlips.preview',
+    staffSlipsViewPreviewLabel: 'ui-circulation.settings.staffSlips.view.previewLabel',
+  };
+
+  const getById = (id) => within(screen.getByTestId(id));
+
+  afterEach(() => {
+    Button.mockClear();
+    PreviewModal.mockClear();
+  });
+
+  beforeEach(() => {
+    render(<StaffSlipTemplateContentSection staffSlip={testInitialValues} />);
+  });
+
+  it('render StaffSlipTemplateContentSection', () => {
+    expect(screen.getByText('ui-circulation.settings.staffSlips.display')).toBeVisible();
+  });
+
+  it('should render staff slips preview label', () => {
+    expect(getById(testIds.staffSlipsPreviewButton).getByText(labelIds.staffSlipsPreview)).toBeVisible();
+  });
+
+  describe('PreviewModal', () => {
+    it('should render PreviewModal component', () => {
+      expect(PreviewModal).toHaveBeenCalledWith(expect.objectContaining({
+        open: false,
+        header: labelIds.staffSlipsViewPreviewLabel,
+        previewTemplate: testInitialValues.template,
+        previewFormat: mockTokensReducerReturnValue,
+        printable: true,
+      }), {});
+    });
+    it('should open and close PreviewModal component', () => {
+      PreviewModal.mockImplementationOnce(({
+        onClose,
+        'data-testid': testId,
+      }) => (
+        // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+        <div
+          data-testid={testId}
+          onClick={onClose}
+        />
+      ));
+
+      expect(PreviewModal).toHaveBeenLastCalledWith(expect.objectContaining({
+        open: false,
+      }), {});
+
+      fireEvent.click(screen.getByTestId(testIds.staffSlipsPreviewButton));
+
+      expect(PreviewModal).toHaveBeenLastCalledWith(expect.objectContaining({
+        open: true,
+      }), {});
+
+      fireEvent.click(screen.getByTestId(testIds.previewModal));
+
+      expect(PreviewModal).toHaveBeenLastCalledWith(expect.objectContaining({
+        open: false,
+      }), {});
+    });
```

Comment:
Could you add separate test for each `expect`?

Reply:
@artem-blazhko 
Here, preview modal is mocked as "mockImplementationOnce", hence I think it makes sense to have the tests this way.
I see other pages on circulation module to hold the tests in the similar fashion.